stream_alert_cli/test.py

```diff
@@ -104,36 +107,42 @@ def format_record(test_record):
         LOGGER_CLI.info('Invalid data type: %s', type(test_record['data']))
         return
 
+    # Get the template file for this particular service
+    template_path = os.path.join(DIR_TEMPLATES, '{}.json'.format(service))
+    with open(template_path, 'r') as service_template:
+        try:
+            template = json.load(service_template)
+        except ValueError as err:
+            LOGGER_CLI.error('Error loading %s.json: %s', service, err)
+            return
     if service == 's3':
-        pass
-
+        # Set the S3 object key to a random value for testing
+        test_record['key'] = ('{:032X}'.format(random.randrange(16**32)))
+        template['s3']['object']['key'] = test_record['key']
+        template['s3']['bucket']['arn'] = 'arn:aws:s3:::{}'.format(source)
+        template['s3']['bucket']['name'] = source
+
+        # Create the mocked s3 object in the designated bucket with the random key
+        put_mocked_s3_object(source, test_record['key'], data)
     elif service == 'kinesis':
-        kinesis_path = os.path.join(DIR_TEMPLATES, 'kinesis.json')
-        with open(kinesis_path, 'r') as kinesis_template:
-            try:
-                template = json.load(kinesis_template)
-            except ValueError as err:
-                LOGGER_CLI.error('Error loading kinesis.json: %s', err)
-                return
-
         template['kinesis']['data'] = base64.b64encode(data)
         template['eventSourceARN'] = 'arn:aws:kinesis:us-east-1:111222333:stream/{}'.format(source)
-        return template
-
     elif service == 'sns':
+        # TODO implement sns testing
         pass
```

Comment:
didn't you want to raise a `NotImplementedError` here instead of failing silently?